src/edge_op.cpp

```diff
@@ -109,14 +117,75 @@ struct SwappableEdge {
   }
 };
 
-struct SortEntry {
-  int start;
-  int end;
-  size_t index;
-  inline bool operator<(const SortEntry& other) const {
-    return start == other.start ? end < other.end : start < other.start;
+struct FlagStore {
```

Comment:
Perhaps a comment about what this struct is used for? I don't entirely follow from reading it.